### PR TITLE
Use ACE_INET_Addr for RTPS configuration

### PR DESCRIPTION
**Pull request by jrw972**

### CODE REVIEW

**Comment by jrw972** [comment]
This commit will most likely not fix any of the broken IPV6 tests but it should help with tests that are now misconfigured.

File: dds/DCPS/RTPS/RtpsDiscovery.cpp [hunk 1]
```diff
@@ -54,9 +54,13 @@ RtpsDiscoveryConfig::RtpsDiscoveryConfig()
   , dx_(2)
   , ttl_(1)
   , sedp_multicast_(true)
-  , default_multicast_group_("239.255.0.1") /*RTPS v2.1 9.6.1.4.1*/
+  , sedp_local_address_(static_cast<u_short>(0), "0.0.0.0")
```
Comment (mitza-oci) [inline dds/DCPS/RTPS/RtpsDiscovery.cpp:57]: There's a lot of these `static_cast`s here.  Consider one of these options:

- named constant `PORT_ZERO` (or similar) at file scope (anonymous namespace)
- functional cast expression `u_short(0)`
- helper function `ACE_INET_Addr fn(const char*)`

File: dds/DCPS/RTPS/Spdp.cpp [hunk 2]
```diff
@@ -2669,11 +2625,14 @@ Spdp::SpdpTransport::join_multicast_group(const DCPS::NetworkInterface& nic,
...
+      ACE_TCHAR buff[256];
+      multicast_address_.addr_to_string(buff, 256);
       ACE_ERROR((LM_ERROR,
                  ACE_TEXT("(%P|%t) ERROR: Spdp::SpdpTransport::join_multicast_group() - ")
-                 ACE_TEXT("failed to join multicast group %C:%hu on %s: %p\n"),
-                 multicast_address_str_.c_str(), mc_port_,
-                 all_interfaces ? ACE_TEXT("all interfaces") : ACE_TEXT_CHAR_TO_TCHAR(nic.name().c_str()), ACE_TEXT("ACE_SOCK_Dgram_Mcast::join")));
+                 ACE_TEXT("failed to join multicast group %s on %s: %p\n"),
+                 buff,
+                 all_interfaces ? ACE_TEXT("all interfaces") : ACE_TEXT_CHAR_TO_TCHAR(nic.name().c_str()),
```
Comment (mitza-oci) [inline dds/DCPS/RTPS/Spdp.cpp:2634]:
```suggestion
                 all_interfaces ? "all interfaces" : nic.name().c_str(),
```
with `%C`

File: dds/DCPS/RTPS/Spdp.cpp [hunk 3]
```diff
@@ -2715,19 +2674,22 @@ Spdp::SpdpTransport::leave_multicast_group(const DCPS::NetworkInterface& nic)
...
     }
 
     if (0 != multicast_socket_.leave(multicast_address_, ACE_TEXT_CHAR_TO_TCHAR(nic.name().c_str()))) {
+      ACE_TCHAR buff[256];
+      multicast_address_.addr_to_string(buff, 256);
       ACE_ERROR((LM_ERROR,
                  ACE_TEXT("(%P|%t) ERROR: Spdp::SpdpTransport::leave_multicast_group() - ")
-                 ACE_TEXT("failed to leave multicast group %C:%hu %p\n"),
-                 multicast_address_str_.c_str(), mc_port_, ACE_TEXT("ACE_SOCK_Dgram_Mcast::leave")));
+                 ACE_TEXT("failed to leave multicast group %s on %s: %p\n"),
```
Comment (mitza-oci) [inline dds/DCPS/RTPS/Spdp.cpp:2691]:
```suggestion
                 ACE_TEXT("failed to leave multicast group %s on %C: %p\n"),
```

**Comment by mitza-oci** [comment]
CI issues summary:
- Java bindings need to get updated
- Inverse dependency Sedp -> RtpsUdpInst (see comment on line 364 of Sedp.cpp on master)

File: dds/DCPS/RTPS/Spdp.cpp [hunk 4]
```diff
@@ -2650,11 +2605,12 @@ Spdp::SpdpTransport::join_multicast_group(const DCPS::NetworkInterface& nic,
...
     if (DCPS::DCPS_debug_level > 3) {
+      ACE_TCHAR buff[256];
+      multicast_address_.addr_to_string(buff, 256);
       ACE_DEBUG((LM_INFO,
                  ACE_TEXT("(%P|%t) Spdp::SpdpTransport::join_multicast_group ")
-                 ACE_TEXT("joining group %C:%hu on %C\n"),
-                 multicast_address_str_.c_str(),
-                 mc_port_,
+                 ACE_TEXT("joining group %s on %C\n"),
+                 all_interfaces ? "all interfaces" : nic.name().c_str(),
```
Comment (mitza-oci) [inline dds/DCPS/RTPS/Spdp.cpp:2613]: Looks like this line should be using `buff` and the next line is the interface name or "all interfaces".

**Comment by mitza-oci** [comment]
Still has linker errors in CI builds

**Comment by mitza-oci** [comment]
New test failures:
```
2020-05-04 22:36:56.027@LM_ERROR@(119324|119324) ERROR: RtpsDiscovery::Config::discovery_config(): failed to parse InteropMulticastOverride 239.255.0.2
2020-05-04 22:36:56.027@LM_ERROR@(119324|119324) ERROR: Service_Participant::load_configuration load_discovery_configuration() returned -1
2020-05-04 22:36:56.027@LM_ERROR@(119324|119324) ERROR: Service_Participant::get_domain_participant_factory: load_configuration() failed.
```

**Pull request merged**